packages/block-library/src/navigation-link/edit.js

```diff
@@ -311,23 +303,35 @@ export default compose( [
 			getClientIdsOfDescendants,
 			hasSelectedInnerBlock,
 			getBlockParentsByBlockName,
+			getColorObjectByColorSlug,
 		} = select( 'core/block-editor' );
 		const { clientId } = ownProps;
 		const rootBlock = head(
 			getBlockParentsByBlockName( clientId, 'core/navigation' )
 		);
-		const navigationBlockAttributes = getBlockAttributes( rootBlock );
+		const navBlockAttrs = getBlockAttributes( rootBlock );
 		const hasDescendants = !! getClientIdsOfDescendants( [ clientId ] )
 			.length;
 		const showSubmenuIcon =
-			!! navigationBlockAttributes.showSubmenuIcon && hasDescendants;
+			!! navBlockAttrs.showSubmenuIcon && hasDescendants;
 		const isParentOfSelectedBlock = hasSelectedInnerBlock( clientId, true );
 
 		return {
 			isParentOfSelectedBlock,
 			hasDescendants,
 			showSubmenuIcon,
-			navigationBlockAttributes,
+			textColor: navBlockAttrs.textColor,
+			backgroundColor: navBlockAttrs.backgroundColor,
+			rgbTextColor: get(
```

Comment:
Should we name the property customTextColor?

Reply:
I'm not 100% sure on this because the `custom…` prefix is already established as a key for the hex color value but ONLY when color is not a swatch. If the selected color is a named swatch, the `custom…` will be undefined. This `rgb` however ALWAYS holds a value, regardless of its source. Does it make sense?

Reply:
yes, also the `custom${ colorName }` is automatically handled by the useColors hook, and it could get confusing if we define the same name structure for this arbitrary property.